feat: allow setting container memory & cpu from task memory & cpu

marcoceppi opened the pull request:
## Description
With the addition of #162 if a user includes extra container definitions and those definitions require CPU/Memory allocations the deployment will fail as the atlantis container definition takes the entire task cpu and memory allocation.

## Motivation and Context
This allows a user to optionally fine tune the allocations of cpu and memory for the atlantis container definition. Most users won't require this feature, unless they're using `extra_container_definitions`.

fixes #168 

## Breaking Changes
This does not break compatibility as by default the values are `null` and if `null` they will default to the `ecs_task_*` definitions, as it was before this change.

## How Has This Been Tested?

```hcl
module "datadog_fargate_container_definition" {
  source  = "cloudposse/ecs-container-definition/aws"
  version = "v0.41.0"

  container_name  = "datadog-agent"
  container_image = "datadog/agent:latest"

  container_memory_reservation = 256
  container_cpu                = 10

  essential = true

  environment = [
    {
      name  = "ECS_FARGATE"
      value = "true"
    }
  ]
}

module "atlantis" {
  source = "github.com/stacklet/terraform-aws-atlantis?ref=issue-168"

  name = "atlantis"

  # DNS
  route53_zone_name = "example.com"

  # Atlantis
  extra_container_definitions = [module.datadog_fargate_container_definition.json_map_object]
  atlantis_github_user           = "marcoceppi"
  atlantis_github_user_token     = data.aws_kms_secrets.atlantis.plaintext["github_user_token"]

  atlantis_hide_prev_plan_comments = "true"

  container_cpu    = 502
  container_memory = 768

  ecs_task_cpu    = 512
  ecs_task_memory = 1024

  allow_repo_config = "true"
}
```

bryantbiggs commented:
hi @marcoceppi - thanks for the PR. could you update the `github-complete` example project to reflect the changes here so that we can both capture how they are utilized as well as test the changes. thank you!

marcoceppi commented:
Can and will do!

marcoceppi commented:
@bryantbiggs I've rebased and updated github-complete with the new values

bryantbiggs commented on variables.tf:
```diff
@@ -299,6 +299,18 @@ variable "ecs_task_memory" {
 }
 
+variable "container_cpu" {
+  description = "The number of cpu units used by the atlantis container. If not specified ecs\_task\_cpu will be used"
```
we don't need to escape here and for some reason it makes TF unhappy!
```suggestion
  description = "The number of cpu units used by the atlantis container. If not specified ecs_task_cpu will be used"
```

bryantbiggs commented on variables.tf:
```diff
@@ -299,6 +299,18 @@ variable "ecs_task_memory" {
+}
+
+variable "container_memory" {
+  description = "The amount (in MiB) of memory used by the atlantis container. If not specified ecs\_task\_memory will be used"
```
```suggestion
  description = "The amount (in MiB) of memory used by the atlantis container. If not specified ecs_task_memory will be used"
```

bryantbiggs reviewed (approved):
once the two changes to the escaping in the variable names are corrected, this should be all set. validated using `examples/github-complete` @antonbabenko 👍🏼

marcoceppi commented:
Thanks for the review @bryantbiggs I've updated the description strings and rebased on the latest in master and fixed the errors from pre-commit 0.14.7 check

bryantbiggs reviewed (approved):
looks good to me - @antonbabenko 👍🏼

antonbabenko commented:
Thanks @marcoceppi !

[v2.33.0](https://github.com/terraform-aws-modules/terraform-aws-atlantis/releases/v2.33.0) has been just released.
